src/app/services/article.service.ts

```diff
@@ -53,28 +53,22 @@ export class ArticleService {
     docRef.set({ path: fullPath, downloadUrl: url });
   }
 
-  async getLatestArticles() {
-    const articlesRef = this.fsdb.collection('articleData/articles/articles');
-    const querySnap = await articlesRef.orderBy('timestamp', 'desc').limit(12).get();
-    return this.arrayFromCollectionSnapshot(querySnap);
+  async latestArticlesRef() {
+    return this.fsdb.collection('articleData/articles/articles', ref => ref.orderBy('timestamp', 'desc').limit(12)).valueChanges();
```

Comment:
And just return the reference instead of the observable. NOTE: This also allows more flexible use in case we wanted to use .snapshotChanges() in a component or another service method.
```suggestion
    return this.fsdb.collection('articleData/articles/articles', ref => ref.orderBy('timestamp', 'desc').limit(12));
```